malfeasance/handler.go

```diff
@@ -31,37 +29,51 @@ var (
 	errInvalidProof  = fmt.Errorf("%w: invalid proof", pubsub.ErrValidationReject)
 )
 
+type (
+	MalfeasanceHandlerV1 func(ctx context.Context, data scale.Type) (types.NodeID, []string, error)
+	MalfeasanceHandlerV2 func(ctx context.Context, data []byte) (types.NodeID, []string, error)
+)
```

Comment:
I don't like to return the labels for the metrics as part of the validation and will change that, besides that the PR however is ready for review.

Reply:
What's your plan for it?

Reply:
I made the callbacks interfaces with a `Validate` method (previous callback) and two methods to report a valid and invalid proof.